src/main/java/htsjdk/samtools/PublicDiskBasedBAMFileIndex.java

```diff
@@ -0,0 +1,18 @@
+package htsjdk.samtools;
+
+import htsjdk.samtools.seekablestream.SeekableStream;
+
+/**
+ NOTE!!!!
+ This is really smelly code!!!!
+
+ This class is not public in HTSJDK. We have opened a PR
+ to make the class public upstream. Once that is merged and
+ a new release of HTSJDK is cut, this file should be removed.
+ */
+public class PublicDiskBasedBAMFileIndex extends DiskBasedBAMFileIndex {
```

Comment:
Looking at https://github.com/samtools/htsjdk/commit/23eb1f02090a507ba3510f20d8b23e94a9e47516 it seems the base class is now public so maybe this is not needed anymore?